pkg/kubernetes/provider.go

```diff
@@ -63,9 +63,19 @@ func getProviderNodeAffinity(provider string, providerList map[string]struct{})
 	if provider == "" || providerList == nil || len(providerList) == 0 {
 		return nil
 	}
-	// if only the default provider exists, there should be no affinity override
-	if provider == DefaultProvider && len(providerList) == 1 {
-		return nil
+
+	if len(providerList) == 1 {
+		_, ok := providerList[provider]
+		if ok {
+			switch provider {
+			case DefaultProvider:
+				// if only the default provider exists, there should be no affinity override
+				return nil
+			case TalosProvider:
+				// if only the Talos provider exists, there should be no affinity override.
+				return nil
```

Comment:
Talos, by default, doesn't seem to apply a label to the nodes that we can use like Google COS does. If you have a cluster with both Talos Linux nodes and other nodes, I don't have a good solution for how to handle that right now.